packages/lisk-transactions/src/transactions/0_transfer.ts

```diff
@@ -0,0 +1,480 @@
+	) {
+		throw new Error('recipientId does not match recipientPublicKey.');
+	}
+
+	if (data && data.length > 0) {
+		if (typeof data !== 'string') {
+			throw new Error(
+				'Invalid encoding in transaction data. Data must be utf-8 encoded string.',
+			);
+		}
+		if (data.length > BYTESIZES.DATA) {
+			throw new Error('Transaction data field cannot exceed 64 bytes.');
+		}
+	}
+};
+
+export class TransferTransaction extends BaseTransaction {
+	public readonly containsUniqueData: boolean;
+	public readonly asset: TransferAsset;
+	public readonly fee: BigNum;
+
+	public constructor(tx: TransactionJSON) {
+		super(tx);
+		const typeValid = validator.validate(transferAssetTypeSchema, tx.asset);
+		const errors = validator.errors
+			? validator.errors.map(
+					error =>
+						new TransactionError(
+							`'${error.dataPath}' ${error.message}`,
+							tx.id,
+							error.dataPath,
+						),
+			  )
+			: [];
+		if (!typeValid || errors.length > 0) {
+			throw new TransactionMultiError('Invalid asset types', tx.id, errors);
+		}
+		this.containsUniqueData = false;
+		this.fee = calculateFee(this.type);
+		this.asset = tx.asset as TransferAsset;
+	}
+
+	public static create(input: TransferInput): object {
+		validateInputs(input);
+		const {
+			amount,
+			recipientId: recipientIdInput,
+			recipientPublicKey,
+			data,
+			passphrase,
+			secondPassphrase,
+		} = input;
+
+		const recipientIdFromPublicKey = recipientPublicKey
+			? getAddressFromPublicKey(recipientPublicKey)
+			: undefined;
+
+		const recipientId = recipientIdInput
+			? recipientIdInput
+			: recipientIdFromPublicKey;
+
+		const transaction = {
+			...createBaseTransaction(input),
+			type: 0,
+			amount,
+			recipientId,
+			recipientPublicKey,
+			fee: calculateFee(0).toString(),
+			asset: data ? { data } : {},
+		};
+
+		if (!passphrase) {
+			return transaction;
+		}
+
+		const transactionWithSenderInfo = {
+			...transaction,
+			// SenderId and SenderPublicKey are expected to be exist from base transaction
+			recipientId: recipientId as string,
+			senderId: transaction.senderId as string,
+			senderPublicKey: transaction.senderPublicKey as string,
+		};
+
+		const transferTransaction = new TransferTransaction(
+			transactionWithSenderInfo,
+		);
+		transferTransaction.sign(passphrase, secondPassphrase);
+
+		return transferTransaction.toJSON();
+	}
+
+	public static fromJSON(tx: TransactionJSON): TransferTransaction {
+		const transaction = new TransferTransaction(tx);
+		const { errors, status } = transaction.validateSchema();
+
+		if (status === Status.FAIL && errors.length !== 0) {
+			throw new TransactionMultiError(
+				'Failed to validate schema',
+				tx.id,
+				errors,
+			);
+		}
+
+		return transaction;
+	}
+
+	protected getAssetBytes(): Buffer {
+		const { data } = this.asset;
+
+		return data ? Buffer.from(data, 'utf8') : Buffer.alloc(0);
+	}
+
+	public assetToJSON(): TransferAsset {
+		return {
+			data: this.asset.data,
```

Comment:
no need of serializeation here so maybe 
```
return {
   ...this.asset,
}
```

Reply:
Also, `TransferAsset` is true in this case, but if we need serialization, it wouldn't be true, so maybe set as `object` here?
Also, maybe we should add default assetToJSON, and override them...

Reply:
not sure if can set a default `assetToJSON` in base transaction since asset field does not exist there.